src/main/scala/gpp/ui/forms/EnumViewSelectBase.scala

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2016-2020 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package lucuma.ui.forms
+
+import cats.syntax.all._
+import crystal.ViewF
+import japgolly.scalajs.react._
+import japgolly.scalajs.react.raw.JsNumber
+import japgolly.scalajs.react.vdom.html_<^._
+import lucuma.core.syntax.all._
+import lucuma.core.util.Display
+import lucuma.core.util.Enumerated
+import react.common._
+import react.semanticui._
+import react.semanticui.collections.form.FormDropdown
+import react.semanticui.collections.form.FormSelect
+import react.semanticui.elements.icon.Icon
+import react.semanticui.elements.label.Label
+import react.semanticui.modules.dropdown.Dropdown._
+import react.semanticui.modules.dropdown._
+
+import scalajs.js
+import scalajs.js.|
+
+/**
+ * Produces a dropdown menu, similar to a combobox.
+ * This is a base trait for various styles.
+ */
+trait EnumViewSelectBase {
+  type FF[_]
+  type AA
+  type BB
```

Comment:
Nice trick. You avoid the need for `Props[Any]` by defining types.

Is it possible to have a `GG[_]` instead of `BB`? For `EnumViewSelect` it can be `cats.Id`.

Reply:
It turns out I can. I will include that in the next commit.